### PR TITLE
Improve DNS lookup debugging

### PR DESCRIPTION
**Pull request by SuperQ**
Rework DNS lookups in order to improve debugging
* Create a unified reverse DNS lookup function.
* Add lookup timing logging
* Add a scrape ID to the logger to identify concurrent scrapes.

### CODE REVIEW

**Comment by SuperQ** [comment]
This should help with debugging https://github.com/SuperQ/chrony_exporter/issues/69

File: collector/collector.go [hunk 1]
```diff
@@ -152,26 +162,45 @@ func (e Exporter) Collect(ch chan<- prometheus.Metric) {
...
+
+func (e Exporter) dnsLookup(logger log.Logger, address net.IP) string {
+	start := time.Now()
+	defer func() {
+		level.Debug(logger).Log("msg", "DNS lookup took", "seconds", time.Since(start).Seconds())
+	}()
+	if !e.dnsLookups {
+		return address.String()
+	}
+	names, err := net.LookupAddr(address.String())
```
Comment (diegocr) [inline collector/collector.go:197]: How about adding some caching capability here?

This is flooding our local DNS Server with several PTR requests every minute 😅

Comment (SuperQ) [reply collector/collector.go:197]: Caching is something best handled by local system design. Something like CoreDNS can handle tens of thousands of requests per second. So, "several PTR requests every minute" is no problem.

Comment (diegocr) [reply collector/collector.go:197]: Yeah, the number of requests is no problem, the problem is that the amount of requests is _cluttering_ the local server _logs_, making hard trying to keep track of the requests we care about.

Thanks for the fast reply, btw.

**Pull request merged**